cli/src/commands/fix.rs

```diff
@@ -185,39 +191,45 @@ pub(crate) fn cmd_fix(
             }
         }
 
-        // Also fix any new paths that were changed in this commit.
+        // There are four behaviors for the combinations of specifying or omitting the
+        // --include-unchanged-files flag or file arguments. The flag determines whether
+        // we diff the revision against its parents or just fix the entire tree. In
+        // either case, the file arguments filter that set of files.
         let tree = commit.tree()?;
-        let parent_tree = commit.parent_tree(tx.repo())?;
-        // TODO: handle copy tracking
-        let mut diff_stream = parent_tree.diff_stream(&tree, &matcher);
-        async {
-            while let Some(TreeDiffEntry {
-                path: repo_path,
-                values,
-            }) = diff_stream.next().await
-            {
-                let (_before, after) = values?;
-                // Deleted files have no file content to fix, and they have no terms in `after`,
-                // so we don't add any tool inputs for them. Conflicted files produce one tool
-                // input for each side of the conflict.
-                for term in after.into_iter().flatten() {
-                    // We currently only support fixing the content of normal files, so we skip
-                    // directories and symlinks, and we ignore the executable bit.
-                    if let TreeValue::File { id, executable: _ } = term {
-                        // TODO: Skip the file if its content is larger than some configured size,
-                        // preferably without actually reading it yet.
-                        let tool_input = ToolInput {
-                            file_id: id.clone(),
-                            repo_path: repo_path.clone(),
-                        };
-                        unique_tool_inputs.insert(tool_input.clone());
-                        paths.insert(repo_path.clone());
-                    }
+        let tree_values: Vec<(RepoPathBuf, Merge<Option<TreeValue>>)> =
+            if args.include_unchanged_files {
+                MergedTree::resolved(Tree::empty(tx.repo().store().clone(), RepoPathBuf::root()))
+            } else {
+                commit.parent_tree(tx.repo())?
+            }
+            .diff_stream(&tree, &matcher)
```

Comment:
nit: I don't have preference about async use, but I think the actual diff here is just:
```rust
let parent_tree = if include_unchanged_files {
    MergedTree::resolved(Tree::empty(..))
} else {
    commit.parent_tree(tx.repo())?
};
```

Reply:
Yes, I was also wondering about the other changes. It seems they should at least go in a separate commit with some justification. Maybe ideally a separate PR so it doesn't have to block the feature change.